src/content/platform-integration/android/compose-activity.md

```diff
@@ -0,0 +1,281 @@
+       // END COPYING HERE
+
+       <activity android:name=".MainActivity" …></activity>
+      …
+</manifest>
+```
+
+The fourth and final code requiring modifications is
+`android/app/src/main/kotlin/com/example/flutter_android_activity/MainActivity.kt`.
+This is where you will begin writing Kotlin code for the Android functionality
+you wish to invoke.
+
+1. Modify the generated `MainActivity` class by adding a `CHANNEL` field and
+`configureFlutterEngine` method:
+
+```kotlin
+class MainActivity: FlutterActivity() {
+
+    // This value must match the `MethodChannel` name in your Dart code.
+    private val CHANNEL = "com.example.flutter_android_activity"
+
+    override fun configureFlutterEngine(flutterEngine: FlutterEngine) {
+        GeneratedPluginRegistrant.registerWith(flutterEngine)
+
+        MethodChannel(flutterEngine.dartExecutor.binaryMessenger, CHANNEL).setMethodCallHandler {
+            call: MethodCall, result: MethodChannel.Result ->
+                when (call.method) {
+                    // Note: This must match the first parameter passed to `platformMethodChannel.invokeMethod`
+                    // in your Dart code.
+                    "launchActivity" -> {
+                        try{
+                            val message = call.arguments  // Takes an object, in this case it's a String
+                            val intent = Intent(this@MainActivity, SecondActivity::class.java)
+                            intent.putExtra("message", message.toString())
+                            startActivity(intent)
+                        } catch (e: Exception){}
+                            result.success(true)
+                        }
+                        else -> {}
+                }
+        }
+    }
+}
+```
+
+2. Add a second Activity to the bottom of the file, which you referenced in
+the previous changes to `AndroidManifest.xml`:
+
+```kotlin
+class SecondActivity : ComponentActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContent {
+                    Surface(modifier = Modifier.fillMaxSize(), color = MaterialTheme.colorScheme.background) {
+                        Column {
+                            Text(text = "Second Activity")
+                            // Note: This must match the shape of the data passed from your Dart code.
+                            Text("" + getIntent()?.getExtras()?.getString("message"))
+                            Button(onClick = {  finish() }) {
+                                Text("Exit")
+                            }
+                        }
+                    }
+        }
+    }
+}
+```
+
+3. Add the necessary imports at the top of the file:
+
+:::note
+Your imports may vary if library versions have changed, or when you write your
+own Kotlin code. Follow your IDE's hints for the correct imports you require.
+:::
+
+```kotlin
+package com.example.flutter_android_activity
```

Comment:
I would have the student add these to the file first and not last. 

Adding it last means that the code is lit up with red squigglies. Might be discouraging if someone is trying to implement as they go.

Moving them to the top could give you the opportunity to discuss in brief the three basic areas of the imports, those needed for the activity and layout, the flutter "compatibility layer" stuff, and then the MethodCall/MethodChannel stuff.

Reply:
I added a paragraph a the top toward this effect - let me know if you think any specific narrative or stage-setting is missing!

Reply:
LGTM.